app/src/main/java/com/xinto/opencord/store/Event.kt

```diff
@@ -0,0 +1,21 @@
+package com.xinto.opencord.store
+
+// TODO: Partialable<T> once DomainMessagePartial can preserve inheritance
+sealed interface Event<T> {
+    data class Add<T>(val data: T) : Event<T>
+    data class Update<T>(val data: T) : Event<T>
+    data class Remove<T>(val id: Long) : Event<T>
```

Comment:
wouldn't remove also send the T?

Reply:
no it wouldn't
MessageDelete for example, sends only id, guildid, channelid which is already a problem but events are a problem for later tbh

Reply:
sure